tests/ModelContextProtocol.Tests/Client/McpClientExtensionsTests.cs

```diff
@@ -37,6 +40,152 @@ public McpClientExtensionsTests(ITestOutputHelper outputHelper)
         _serverTask = server.RunAsync(cancellationToken: _cts.Token);
     }
 
+    [Theory]
+    [InlineData(null, null)]
+    [InlineData(0.7f, 50)]
+    [InlineData(1.0f, 100)]
+    public async Task CreateSamplingHandler_ShouldHandleTextMessages(float? temperature, int? maxTokens)
+    {
+        // Arrange
+        var mockChatClient = new Mock<IChatClient>();
+        var requestParams = new CreateMessageRequestParams
+        {
+            Messages =
+            [
+                new SamplingMessage
+                {
+                    Role = Role.User,
+                    Content = new Content { Type = "text", Text = "Hello" }
+                }
+            ],
+            Temperature = temperature,
+            MaxTokens = maxTokens
+        };
+
+        var cancellationToken = CancellationToken.None;
+        var expectedResponse = new ChatResponse
+        {
+            Messages = { new ChatMessage { Role = ChatRole.Assistant, Contents = { new TextContent("Hi there!") } } },
+            ModelId = "test-model",
+            FinishReason = ChatFinishReason.Stop
+        };
+
+        mockChatClient
+            .Setup(client => client.GetResponseAsync(It.IsAny<IList<ChatMessage>>(), It.IsAny<ChatOptions>(), cancellationToken))
+            .ReturnsAsync(expectedResponse);
+
+        var handler = McpClientExtensions.CreateSamplingHandler(mockChatClient.Object);
```

Comment:
The signature of this delegate was updated. I expect this will need to be tweaked.

Reply:
Updated. Please take a look @stephentoub @eiriktsarpalis